test/shopify-cli/core/monorail_test.rb

```diff
@@ -121,6 +121,52 @@ def test_log_event_handles_errors
         end
       end
 
+      def test_log_script_event_does_not_have_api_key_in_payload
+        enabled_and_consented(true, true)
+        Timecop.freeze do |time|
+          this_time = (time.utc.to_f * 1000).to_i
+          stub_request(:post, Monorail::ENDPOINT_URI)
```

Comment:
I realise this is following the existing approach from previous tests in this file, but it seem rather verbose (and fragile). I wonder if we can do anything to improve the overall approach here?

(see [Obscure Test Smell](https://learning.oreilly.com/library/view/xUnit+Test+Patterns:+Refactoring+Test+Code/9780131495050/ch15.html#ch15lev1sec1) in xUnit Test Patterns).

Reply:
Even though my comment above makes this test case unnecessary, I still agree with this suggestion for future consideration!